packages/trpc/server/routers/viewer/bookings/get.handler.ts

```diff
@@ -317,13 +318,69 @@ async function getBookings({
               },
             },
           },
+          {
+            eventType: {
+              parent: {
+                team: {
+                  members: {
+                    some: {
+                      userId: user.id,
+                      role: {
+                        in: ["ADMIN", "OWNER"],
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
         ],
         AND: [passedBookingsStatusFilter, ...filtersCombined],
       },
       orderBy,
       take: take + 1,
       skip,
     }),
+    // only get personal bookings of current user's team members
+    prisma.booking.findMany({
```

Comment:
Thank you for your contribution 🙏 @SomayChauhan 

Just as a postmortem. 

The addition of this specific query it is causing the DB CPU to be at 100%. Causing connection issues in the whole app. We will have to revert this PR in order fix it.

My main guess is that this query is missing a select statement. Causing to query ALL data from the booking.

My other guess is that this Promise.all is reaching its limit. While it might indeed be "faster" to fetch these all at once. It is also true that this will causing CPU spikes as we seen today in the v3.8.8 release.

Thanks again for the patience 🙏🏽